src/gurobi_ml/sklearn/logistic_regression.py

```diff
@@ -192,39 +205,91 @@ def default_pwl_attributes() -> dict:
             }
         return {"FuncNonlinear": 1}
 
+    def _addGenConstrIndicatorMvarV10(self, binvar, binval, lhs, sense, rhs, name):
+        """This function is to work around the lack of MVar compatibility in
+        Gurobi v10 indicator constraints.  Note, it is not as flexible as Model.addGenConstrIndicator
+        in V11+.  If support for v10 is dropped this function can be removed.
+
+        Parameters
+        ----------
+        binvar : MVar
+        binval : {0,1}
+        lhs : MVar or MLinExpr
+        sense : (char)
+            Options are gp.GRB.LESS_EQUAL, gp.GRB.EQUAL, or gp.GRB.GREATER_EQUAL
+        rhs : scalar
+        name : string
+        """
+        assert binvar.shape == lhs.shape
+        total_constraints = np.prod(binvar.shape)
+        binvar = binvar.reshape(total_constraints).tolist()
+        lhs = lhs.reshape(total_constraints).tolist()
+        for index in range(total_constraints):
+            self.gp_model.addGenConstrIndicator(
+                binvar[index],
+                binval,
+                lhs[index],
+                sense,
+                rhs,
+                name=self._indexed_name(index, name),
+            )
+
     def _mip_model(self, **kwargs):
         """Add the prediction constraints to Gurobi."""
+        affinevars = self.gp_model.addMVar(
+            self.output.shape, lb=-gp.GRB.INFINITY, name="affine_trans"
+        )
+        self._add_regression_constr(output=affinevars)
+
         if self.output_type == "classification":
             # For classification we need an extra binary variable
-            log_result = self.gp_model.addMVar(
-                self.output.shape, lb=-gp.GRB.INFINITY, name="log_result"
-            )
             bin_output = self.gp_model.addMVar(
                 self.output.shape, vtype=gp.GRB.BINARY, name="bin_output"
             )
 
-            self.gp_model.addConstr(bin_output >= log_result - 0.5 + self.epsilon)
-            self.gp_model.addConstr(bin_output <= log_result + 0.5)
+            # Workaround for MVars in indicator constraints for v10.
```

Comment:
It's fine but note that in other places I actually didn't update to v11 syntax not to loose compatibility.
E.g.: for decision tree I just do a v10 compatible loop:

https://github.com/Gurobi/gurobi-machinelearning/blob/66fb44a1528956b799fdb46bd140ab90073d7a02/src/gurobi_ml/modeling/decision_tree/decision_tree_model.py#L114-L122

The v11 code should be faster. So it's fine like this is but you could have just kept the slow loop.

Reply:
I figured once we stop supporting v10 that all subsequent releases of gurobi-machinelearning would to, at which point we can dump this function I've added and it will be very obvious what to do for any future maintainer.